internal/app/machined/internal/phase/network/network.go

```diff
@@ -60,11 +52,15 @@ func (task *InitialNetworkSetup) runtime(r runtime.Runtime) (err error) {
 	if r.Config() != nil {
 		log.Println("merging user defined network configuration")
 
-		if err = netconf.BuildOptions(r.Config()); err != nil {
+		if err = netconf.BuildOptionsFromConfig(r.Config()); err != nil {
 			log.Fatal(err)
 		}
 	}
 
+	if err = netconf.BuildOptionsFromKernel(); err != nil {
```

Comment:
This order still strikes me as wrong:  options from config should override options from kernel.